app/mixins/build-favicon.coffee

```diff
@@ -0,0 +1,36 @@
+`import Ember from 'ember'`
+`import { colorForState } from 'travis/utils/helpers'`
+`import FaviconManager from 'travis/utils/favicon-manager'`
+
+Mixin = Ember.Mixin.create
+  actions:
+    faviconStateDidChange: (state) ->
+      if state
+        @setFaviconForState(state)
+      else
+        @setDefault()
+
+  init: ->
+    @faviconManager = new FaviconManager()
+
+    @_super.apply this, arguments
+
+  setFaviconForState: (state) ->
+    color = colorForState(state)
+
+    images = {
+      red: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP4z8DwHwAFAAH/VscvDQAAAABJRU5ErkJggg==',
+      yellow: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP4/5ThPwAHrwLktvrIVgAAAABJRU5ErkJggg==',
+      green: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mOwv276HwAE7wJLzZ3V0gAAAABJRU5ErkJggg==',
+      gray: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mMoqyj7DwAFMAJkMSbkxgAAAABJRU5ErkJggg=='
+    }
+
+    @setFavicon(images[color])
+
+  setDefault: ->
```

Comment:
Changing url doesn't work as good as setting a data uri (it was sometimes changing and sometimes stayed unchanged when I was testing on chrome). I'll check if we can generate data-uris based on images, it should be easy with the tool we use for CSS already.